src/process.rs

```diff
@@ -112,16 +118,31 @@ impl PtyProcess {
                 dup2(slave_fd, STDOUT_FILENO)?;
                 dup2(slave_fd, STDERR_FILENO)?;
 
+                unsafe {
+                    match ioctl(master_fd.as_raw_fd(), TIOCSCTTY) {
+                        0 => Ok(()),
+                        _ => Err(nix::Error::last()),
+                    }?;
+                }
```

Comment:
I assume the unsafe block is only needed for the `ioctl` call here, isn't it?
also, we don't need to match here, I think.

```suggestion
                if 0 != unsafe { ioctl(master_fd.as_raw_fd(), TIOCSCTTY) } {
                    return Err(nix::Error::last())
                }
```